src/main/java/uk/gov/service/notify/NotificationClient.java

```diff
@@ -312,6 +337,8 @@ public static JSONObject prepareUpload(final byte[] documentContents, boolean is
         JSONObject jsonFileObject = new JSONObject();
         jsonFileObject.put("file", fileContent);
         jsonFileObject.put("is_csv", isCsv);
+        jsonFileObject.put("confirm_email_before_download", JSONObject.NULL);
+        jsonFileObject.put("retention_period", JSONObject.NULL);
```

Comment:
hmmm i'd like it if we didn't redefine the implementation multiple times - scraping my java memory back from uni.... can this function now just call `prepareUpload(documentContents, isCsv, null, null)` (the same way the shorter function below calls this function with a default value for isCsv)?

if that doesnt work due to type stuff (ie: can you pass in null to a boolean) then thats fine, however in that case i think the docs might need to be tweaked

Reply:
Yep, the types are incompatible:

`[ERROR] /var/project/src/main/java/uk/gov/service/notify/NotificationClient.java:[331,55] incompatible types: <nulltype> cannot be converted to boolean`